lib/oban/queue/executor.ex

```diff
@@ -245,6 +251,7 @@ defmodule Oban.Queue.Executor do
     |> Map.take([:id, :args, :queue, :worker, :attempt, :max_attempts, :tags])
     |> Map.put(:job, job)
     |> Map.put(:prefix, conf.prefix)
+    |> Map.put(:name, conf.name)
```

Comment:
After constantly playing catch-up with exposing various job fields I eventually put the `job` in there directly. The telemetry events happen in the same process, so there isn't any data copying. 

What if we do the same thing with the config? We can put the config in and soft deprecate `prefix` by removing it from the telemetry docs.

Reply:
That sounds good. Updated the code and the tests to reflect that change :+1: